### PR TITLE
feat: created product listing page with filters

### PR DESCRIPTION
**Pull request by nakulsharma15**
Created Product Listing Page Containing filters of the following types:
1. Filter by category
2. Filter by ratings
3. Sort by: Price - (High to Low) or (Low to High)

User can also click on the category on homepage and gets the filtered product respective to that category.

### CODE REVIEW

**Review by Priyanshi-Tiwari — APPROVED** [review]
- The code looks good. 
- good work on breaking things into smaller components.
- clean code and descriptive classnames

File: src/Components/Card.jsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
+import "./Styles/Card.css";
+
+import React from "react";
```
Comment (ttahm3d) [inline src/Components/Card.jsx]: From React version 17, this import is not required.

File: src/Components/Card.jsx [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+                    <img class="card-img flex-justify-center" src={Item.imageUrl} alt="Product" />
+                </div>
+
+                <div class="card-title item-price-detail">
+                    <h3>{Item.name}</h3>
+                    <p>₹{Item.price}  {!(Item.discount === 0) ? <s class="text-s">₹{Item.prevPrice}</s> : <s></s>}</p>
+                </div>
+
+                <div className="discount-badge">
+                    {!(Item.discount === 0) ? <div class="discount special-badge text-overlay">{Item.discount}% off</div> : <div></div>}
```
Comment (ttahm3d) [inline src/Components/Card.jsx]: Dont render empty div's. you can pass `null`

File: src/Filters/FilterFunction.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
+const sortby =({sortBy},array)=>{
+    switch(sortBy){
+        case "HIGHTOLOW":
+            return sortBy==="HIGHTOLOW"?[...array].sort((a,b)=>Number(b.price)- Number(a.price)):[...array];
+        case "LOWTOHIGH":
+            return sortBy==="LOWTOHIGH"?[...array].sort((a,b)=>Number(a.price)- Number(b.price)):[...array];
+        default:
+            return [...array]
```
Comment (ttahm3d) [inline src/Filters/FilterFunction.js]: Since you are not changing anything in default, you can return the array normally without spreading

**Review by ttahm3d — APPROVED** [review]
Great job. Looks good. Have provided suggestions for minor things.

File: src/Components/Card.jsx (hunk 1 above)
Comment (nakulsharma15) [reply src/Components/Card.jsx]: Updated it in the latest commit

File: src/Components/Card.jsx (hunk 2 above)
Comment (nakulsharma15) [reply src/Components/Card.jsx]: Thank you Tahir for pointing this out. Changed it in latest commit.

File: src/Filters/FilterFunction.js (hunk 3 above)
Comment (nakulsharma15) [reply src/Filters/FilterFunction.js]: Changed it in the latest commit.

**Pull request merged**